api/app/Models/PoolCandidate.php

```diff
@@ -675,8 +675,12 @@ public function isDraft()
      */
     public function scopeAuthorizedToView(Builder $query)
     {
-        $userId = Auth::user()->id;
-        $user = User::find($userId);
+        $authUser = Auth::user();
+        if (! $authUser) {
+            return $query->where('id', null);
```

Comment:
I don't think I understand `$query->where('id', null);`.  Is this like a "where true = false" expression?  Why not just throw an exception and abort the query?

Reply:
I'm not sure, I kind of just assumed we would want the query to run on a happy path and just return an empty result. So if the rest of the query is okay it won't crash a page and we are still protecting data.

If we are okay with a panic response, I can throw an exception instead.

Reply:
Yes, exactly, this is a quick way to force the query to return zero results. We do the same thing in `User.scopeAuthorizedToView`. We decided that scopes should not be concerned with doing authorization - if a user isn't permitted to see any results, the correct behaviour for the scope is to return no results. As eric says, that allows other parts of the query to return without crashing.